src/metrics/rules/active_ruleset_bench_test.go

```diff
@@ -0,0 +1,278 @@
+package rules
+
+import (
+	"fmt"
+	"math"
+	"testing"
+	"time"
+
+	"github.com/m3db/m3/src/metrics/aggregation"
+	"github.com/m3db/m3/src/metrics/filters"
+	"github.com/m3db/m3/src/metrics/metric"
+	metricid "github.com/m3db/m3/src/metrics/metric/id"
+	"github.com/m3db/m3/src/metrics/metric/id/m3"
+	"github.com/m3db/m3/src/metrics/pipeline"
+	"github.com/m3db/m3/src/metrics/policy"
+	xtime "github.com/m3db/m3/src/x/time"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var benchmarkCases = []struct {
+	name string
+	id   string
+}{{
+	name: "rollup ID",
+	id:   fmt.Sprintf("m3+%s+m3_rollup=true,rtagName1=val1,rtagName2=val2", newTestRuleName("rollup", 5)),
+}, {
+	name: "mapping ID",
+	id:   "m3+test+rtagName1=val1,rtagName2=val2",
+}}
+
+// Results (2024-10-18, Macbook Pro M1 Max):
+//
+// goos: darwin
+// goarch: arm64
+// pkg: github.com/m3db/m3/src/metrics/rules
+// BenchmarkActiveRuleSet_ReverseMatch
+// BenchmarkActiveRuleSet_ReverseMatch/rollup_ID
+// BenchmarkActiveRuleSet_ReverseMatch/rollup_ID-10         	      21	  52730040 ns/op	43040917 B/op	  360008 allocs/op
+// BenchmarkActiveRuleSet_ReverseMatch/mapping_ID
+// BenchmarkActiveRuleSet_ReverseMatch/mapping_ID-10        	       2	 533871292 ns/op	1089136572 B/op	 4670173 allocs/op
+// PASS
+func BenchmarkActiveRuleSet_ReverseMatch(b *testing.B) {
+	for _, tc := range benchmarkCases {
+		b.Run(tc.name, func(b *testing.B) {
+			b.ReportAllocs()
+
+			id := m3.NewID([]byte(tc.id), nil)
+
+			deps := setupActiveRuleSetBenchmark(b)
+			sanityCheckBenchmark(b, deps, id)
+
+			// N.B.: this is actually a heavy alloc, it turns out.
+			typesOptions := aggregation.NewTypesOptions()
+
+			b.ResetTimer()
+
+			var benchResult MatchResult
+			for i := 0; i < b.N; i++ {
+				var loopResult MatchResult
+				for j := 0; j < 10_000; j++ {
```

Comment:
why do you do this? if you want a certain number of runs, just use the benchmark cli to specify them

```
        -benchtime t
            Run enough iterations of each benchmark to take t, specified
            as a time.Duration (for example, -benchtime 1h30s).
            The default is 1 second (1s).
            The special syntax Nx means to run the benchmark N times
            (for example, -benchtime 100x).

        -count n
            Run each test, benchmark, and fuzz seed n times (default 1).
            If -cpu is set, run n times for each GOMAXPROCS value.
            Examples are always run once. -count does not apply to
            fuzz tests matched by -fuzz.
```

Reply:
Yeah you're totally right; misguided attempt at "realism" I think. Will change.